### PR TITLE
chore: added script for local build

### PR DESCRIPTION
**Pull request by domreuter**
This script will build the different projects and start up the containers

### CODE REVIEW

**Comment by florianrusch-zf** [comment]
Hi @domreuter thanks for you PR! Sadly the steps in the build file are not complete. Before you can run this, you need to build the EDC modules and to publish it into your local maven cache.

For the moment we added the IntelliJ Run Configs into the repository, so everybody can use them independently from their operation systems. If we want to add such a script, we should ensure, that it runs on all operating systems as well.

Additionally we plan to create a better `README.md` to provide a better onboarding (Take a look at #25).

Therefore it would be awesome, if you can ensure, that your script runs at least on Windows, Linux and Mac.

File: api-wrapper/build_run_local.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+./gradlew ${MAVEN_OPTS} clean build
+# change directory to root 
+cd ../..
+
+# 
+export DOCKER_PLATFORM=${DOCKER_PLATFORM:-linux/amd64}
+
+# build docker 
+docker-compose --file docker-compose.yml --project-name catenax_at_home --verbose build
+docker-compose --file docker-compose.yml --project-name catenax_at_home --verbose up
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:54]: - The `--file` flag is not needed.
- The `--verbose` flat also.
- Setting the `--project-name` flag is a great idea! But I would then set in the `docker-compose.yml` files directly. Can you do that?

```suggestion
docker-compose build
docker-compose up
```

File: api-wrapper/build_run_local.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#   Unix, zsh, java 11 (java) and maven (mvn) in the $PATH.
+#
+# Synposis:
+#   ./build_run_local.sh
+#
+# Comments:
+#   
+#
+
+export MAVEN_OPTS=(${GRADLE_PROPS[*]})
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:25]: Do we really need this?

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]

File: api-wrapper/build_run_local.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+# build semantics
+cd services/semantics
+mvn clean install -DskipTests
+# build api-wrapper
+cd ../api-wrapper
+chmod +x gradlew
```
[the remark this answers is not in the document]
Comment (domreuter) [reply api-wrapper/build_run_local.sh:32]: Had some problems with one of the gradle wrappers, it was not marked as executable per default. Need to figure out which one exactly it was

File: api-wrapper/build_run_local.sh (hunk 1 above)
Comment (domreuter) [reply api-wrapper/build_run_local.sh:54]: The `--file` is there for some other reasons. I will work with multiple compose files, therefore i need this one. By now only the default compose-file is referenced here.

I will integrate the project-name in the docker-compose for a better overview of the container naming.

File: api-wrapper/build_run_local.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+./gradlew ${MAVEN_OPTS} clean build
+# build data-plane
+cd ../data-plane
+chmod +x gradlew
+./gradlew ${MAVEN_OPTS} clean build
+# change directory to root 
+cd ../..
+
+# 
+export DOCKER_PLATFORM=${DOCKER_PLATFORM:-linux/amd64}
```
[the remark this answers is not in the document]
Comment (domreuter) [reply api-wrapper/build_run_local.sh]: fixed this!

File: api-wrapper/build_run_local.sh (hunk 2 above)
Comment (domreuter) [reply api-wrapper/build_run_local.sh]: fixed: no need for this anymore!

File: api-wrapper/docker-compose-consumer.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      EDC_FS_CONFIG: /config/configuration.properties
+    volumes:
+      - ./config/api-wrapper.config:/config
+
+  consumer-aas-proxy-service:
+    image: catenax_at_home_consumer-aas-proxy-service:latest
+    platform: ${DOCKER_PLATFORM:-linux/amd64}
+    ports:
+      - "4245:4245" # default
+      - "5014:8090" # debugging
```
Comment (florianrusch-zf) [inline api-wrapper/docker-compose-consumer.yml:50]: Missing empty line at EOF
```suggestion
      - "5014:8090" # debugging

```

File: api-wrapper/docker-compose-provider.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      EDC_FS_CONFIG: /config/configuration.properties
+    volumes:
+      - ./config/provider-data-plane.config:/config
+
+  provider-backend-service:
+    image: catenax_at_home_provider-backend-service:latest
+    platform: ${DOCKER_PLATFORM:-linux/amd64}
+    ports:
+      - "8194:8080" # default
+      - "5010:8090" # debugging
```
Comment (florianrusch-zf) [inline api-wrapper/docker-compose-provider.yml:38]: Missing empty line at EOF

```suggestion
      - "5010:8090" # debugging

```

File: api-wrapper/build_run_local.sh [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+# change directory to root 
+cd ../..
+echo "07: --> DOCKER COMPOSE BUILD AND RUN LOCAL CONTEXT"
+docker-compose --file docker-compose.yml --project-name catenax_at_home --verbose build
+
+docker-compose --file docker-compose-consumer.yml --project-name consumer_catenax_at_home up &
+docker-compose --file docker-compose-provider.yml --project-name provider_catenax_at_home up &
+docker-compose --file docker-compose-centralservices.yml --project-name centralservices_catenax_at_home up &
+
+#docker-compose --file docker-compose.yml --project-name catenax_at_home --verbose up
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:88]: Missing empty line at EOF
```suggestion
#docker-compose --file docker-compose.yml --project-name catenax_at_home --verbose up

```

File: api-wrapper/build_run_local.sh [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+echo "02: --> BUILD SEMANTICS SOURCES"
+cd services/semantics
+./mvnw clean install -DskipTests
+
+#############################
+# Build api-wrapper
+#############################
+echo "03: --> BUILD API-WRAPPER SOURCES"
+cd ../api-wrapper
+./gradlew clean build
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:53]: If we skip the tests in the steps 1 & 2, we should do it in all other steps also
```suggestion
./gradlew clean build -x test
```

File: api-wrapper/build_run_local.sh [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+#############################
+# Build backend-data-service
+#############################
+echo "04: --> BUILD BACKEND-DATA-SERVICE SOURCES"
+cd ../backend-data-service
+./gradlew clean build
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:60]:
```suggestion
./gradlew clean build -x test
```

File: api-wrapper/build_run_local.sh [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+#############################
+# Build control-plane
+#############################
+echo "05: --> BUILD CONTROL-PLANE SOURCES"
+cd ../control-plane
+./gradlew clean build
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:67]:
```suggestion
./gradlew clean build -x test
```

File: api-wrapper/build_run_local.sh [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+#############################
+# Build data-plane
+#############################
+echo "06: --> BUILD DATA-PLANE SOURCES"
+cd ../data-plane
+./gradlew ${MAVEN_OPTS} clean build
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:74]: I think we don't need the `MAVEN_OPTS` anymore or?

```suggestion
./gradlew clean build -x test
```

File: api-wrapper/build_run_local.sh [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+# Docker Compose Build and Run
+##############################
+# change directory to root 
+cd ../..
+echo "07: --> DOCKER COMPOSE BUILD AND RUN LOCAL CONTEXT"
+docker-compose --file docker-compose.yml --project-name catenax_at_home --verbose build
+
+docker-compose --file docker-compose-consumer.yml --project-name consumer_catenax_at_home up &
+docker-compose --file docker-compose-provider.yml --project-name provider_catenax_at_home up &
+docker-compose --file docker-compose-centralservices.yml --project-name centralservices_catenax_at_home up &
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:86]: Didn't we discussed last time that we want to move the `project-name` into the `docker-compose.yml` files?

File: api-wrapper/build_run_local.sh [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+#############################
+# Clone OOS EDC and build sources
+# |-> Milestone 3.1
+# https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/milestone-3.1
+#############################
+echo "01: --> CLONE AND BUILD EDC SOURCES"
+DIR="DataSpaceConnector"
+if [ -d "$DIR" ]; then
+  rm -rf $DIR
+  echo "Folder ${DIR} already exists: Deleting ${DIR}"
```
Comment (florianrusch-zf) [inline api-wrapper/build_run_local.sh:34]:
```suggestion
  echo "Folder ${DIR} already exists: Deleting ${DIR}"
  rm -rf $DIR
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]

File: api-wrapper/build_run_local.sh [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+#
+# Comments:
+#   
+#
+
+export MAVEN_OPTS=(${GRADLE_PROPS[*]})
+
+# build semantics
+cd services/semantics
+mvn clean install -DskipTests
```
[the remark this answers is not in the document]
Comment (domreuter) [reply api-wrapper/build_run_local.sh]: fixed: uses internal mvn wrapper now

File: api-wrapper/build_run_local.sh (hunk 12 above)
Comment (domreuter) [reply api-wrapper/build_run_local.sh:86]: If you found an easy way to do so feel free to add this one, atm it is only possible via environment variables.

Comment (florianrusch-zf) [reply api-wrapper/build_run_local.sh]: With the newest Compose-Version its possible 😉  But I'm not sure if we should enforce all the users to upgrade to the latest version 🤔

File: api-wrapper/docker-compose-provider.yml [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:8090"
+    volumes:
+      - ./config/provider-control-plane.config:/config
+      - ./certs/cert.pfx:/cert.pfx
+
+  provider-data-plane:
+    image: catenax_at_home_provider-data-plane:latest    
+    ports:
+      - "8491:9191" # default (/api)
+      - "8492:9192" # public (/public)
```
Comment (florianrusch-zf) [inline api-wrapper/docker-compose-provider.yml:29]:
```suggestion
      - "8492:9192" # public (/public)
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]

File: api-wrapper/build_run_local.sh (hunk 12 above)
Comment (domreuter) [reply api-wrapper/build_run_local.sh]: any sources about that?

**Pull request closed without merging**